### PR TITLE
Ocular language additions

### PR DESCRIPTION
**Pull request by NWeber-SL**

### CODE REVIEW

File: languages/golang.md [hunk 1]
```diff
@@ -30,17 +30,18 @@ where
...
-After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md) and [authenticating](../using-cli/authenticating.md), use this command to run ShiftLeft Ocular
+After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md), [authenticating](../using-cli/authenticating.md) and [starting ShiftLeft Ocular](../using-ocular/getting-started/starting.md), [create the Code Property Graph (CPG)](../using-ocular/getting-started/create-cpg.md) for your Golang application using
 
 ```scala
-sl ocular
+ocular> createCpg(<inputPaths>)
 ```
 
-### Next Steps
+where `<inputPaths>` is the path of the target application; multiple applications are separated by a comma. For Golang, the path is the package, for example `createCpg("helloshiftleftgo")`.
```
Comment (vlad-alexandru-ionescu) [inline languages/golang.md]: This seems incorrect. Can you help us out with the right formulation here, @Ferada?

File: languages/llvm.md [hunk 2]
```diff
@@ -2,16 +2,19 @@
...
-After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md) and [authenticating](../using-cli/authenticating.md), use this command to run ShiftLeft Ocular
+After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md), [authenticating](../using-cli/authenticating.md) and [starting ShiftLeft Ocular](../using-ocular/getting-started/starting.md), [create the Code Property Graph (CPG)](../using-ocular/getting-started/create-cpg.md) for your LLVM application using
 
 ```scala
-sl ocular
+ocular> createCpg(<inputPaths>)
 ```
 
-## Next Steps
+where `<inputPaths>` is the path of the target application; multiple applications are separated by a comma. For LLVM, the path is the directory containing the LLVM files.
```
Comment (vlad-alexandru-ionescu) [inline languages/llvm.md]: This seems a little vague - can you help us out with the right formulation, @AlexDenisov?

File: languages/golang.md (hunk 1 above)
Comment (vlad-alexandru-ionescu) [reply languages/golang.md]: Also, under the Inspect part within this file (not edited in this PR), it says

>	`<path>` location of the `.go` file to be analyzed.

Which seems vague. Should it be "location of the `.go` files of the main package to be analyzed. Or something along those lines, @Ferada ?

Perhaps we should additionally say this is usually "the same arguments you would pass in to a `go build` command.

Comment (Ferada) [reply languages/golang.md]: > Also, under the Inspect part within this file (not edited in this PR), it says
> 
> > `<path>` location of the `.go` file to be analyzed.
> 
> Which seems vague. Should it be "location of the `.go` files of the main package to be analyzed". Or something along those lines, @Ferada ?

No, you generally do not want to specify individual files, unless you actually just want to process a single file, but that wrecks havoc with imports. Like below, the argument is usually the package name, or a package specifier that might include all of the subprojects, e.g. `./...`.

For Go there might a workaround necessary for the latter at the moment:

```
 createCpg(List("/home/olof/src/go/src/github.com/ShiftLeftSecurity/go-services", "/home/olof/src/go/src/github.com/ShiftLeftSecurity/go-services/..."), "GOLANG")
```

Basically, Ocular requires the paths to exist. Well, at least the first one. Since all argument are passed to the frontend though, we can use the above to make it analyse `go-services`' packages too. Also note the need for `"GOLANG"` in case the heuristic doesn't find anything to determine the language automatically.

> Perhaps we should additionally say this is usually "the same arguments you would pass in to a `go build` command.

Yes. "Usually the same arguments you would pass in to a `go build` command, including any tags, command line flags etc."

Comment (NWeber-SL) [reply languages/golang.md]: Article updated with information. Waiting for Olof to review.

File: languages/llvm.md (hunk 2 above)
Comment (NWeber-SL) [reply languages/llvm.md]: Article updated with information from Alex. Waiting for him to review.

File: languages/golang.md [hunk 3]
```diff
@@ -30,17 +30,18 @@ where
...
-After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md) and [authenticating](../using-cli/authenticating.md), use this command to run ShiftLeft Ocular
+After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md), [authenticating](../using-cli/authenticating.md) and [starting ShiftLeft Ocular](../using-ocular/getting-started/starting.md), [create the Code Property Graph (CPG)](../using-ocular/getting-started/create-cpg.md) for your Golang application using
 
 ```scala
-sl ocular
+ocular> createCpg(<inputPaths> "GOLANG")
 ```
 
-### Next Steps
+where `<inputPaths>` is the path of the target application; multiple applications are separated by a comma. For Golang, the path is the package or a package specifier that includes all of the subprojects, using the same arguements you would pass in  a go build command. For example `createCpg("helloshiftleftgo")`.
```
[the remark this answers is not in the document]
Comment (Ferada) [reply languages/golang.md]: I doubt that analysing multiple applications together makes any sense, considering they won't be calling each other. But then again I've only used Ocular in Go and Java for a bit. It's certainly untested there.

What does make sense is analysing multiple packages of the same application, or including packages that would otherwise have to be covered by annotations instead.

File: languages/java.md [hunk 4]
```diff
@@ -37,17 +37,21 @@ where
...
+
+You create the Code Property Graph (CPG) for your Java application using
 
 ```scala
-sl ocular
+ocular> createCpg(<inputPaths>)
 ```
 
-### Next Steps
+where `<inputPaths>` is the path of the target application; multiple applications are separated by a comma. For Java, the path is the archive (JAR or WAR file). For example, `createCpg("subjects/hello-shiftleft-0.0.1-SNAPSHOT.jar").`
```
Comment (Ferada) [inline languages/java.md]:
```suggestion
where `<inputPaths>` is the path of the target application; multiple applications are separated by a comma. For Java, the path is the archive (JAR, WAR or EAR file). For example, `createCpg("subjects/hello-shiftleft-0.0.1-SNAPSHOT.jar").`
```

File: languages/java.md [hunk 5]
```diff
@@ -37,17 +37,21 @@ where
...
 
-[Create the Code Property Graph (CPG)](../using-ocular/getting-started/create-cpg.md)
+For more information, including additional options, refer to the article [Creating the CPG](../using-ocular/getting-started/create-cpg.md)
```
Comment (Ferada) [inline languages/java.md]:
```suggestion
For more information, including additional options, refer to the article [Creating the CPG](../using-ocular/getting-started/create-cpg.md).
```

File: languages/llvm.md [hunk 6]
```diff
@@ -2,16 +2,27 @@
...
 ShiftLeft Ocular supports the examination and investigation of LLVM applications. LLVM is a compiler infrastructure project providing a set of compiler and toolchain technologies, which can be used to develop a front end for any programming language and a back end for any instruction set architecture.
 
-After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md) and [authenticating](../using-cli/authenticating.md), use this command to run ShiftLeft Ocular
+ShiftLeft Ocular examines LLVM bitcode in the following forms:
+
+* LLVM IR (human-readable representation)
+* LLVM Bitcode (bitstream representation)
+* Embedded Bitcode (bitstream representation embedded into a binary)
+
+Before using ShiftLeft Ocular, you must extract the LLVM bitcode out of high-level source code by introducing the appropriate flags into your build system (i.e. `-emit-llvm`, `-flto`, or `-fembed-bitcode`), or use use `whole-program-llvm`.
```
Comment (Ferada) [inline languages/llvm.md]:
```suggestion
Before using ShiftLeft Ocular, you must extract the LLVM bitcode out of high-level source code by introducing the appropriate flags into your build system (i.e. `-emit-llvm`, `-flto`, or `-fembed-bitcode`), or use `whole-program-llvm`.
```

File: languages/scala.md [hunk 7]
```diff
@@ -31,18 +31,22 @@ where
...
-After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md) and [authenticating](../using-cli/authenticating.md), use this command to run ShiftLeft Ocular
+After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md), [authenticating](../using-cli/authenticating.md) and [starting ShiftLeft Ocular](../using-ocular/getting-started/starting.md), you create the Code Property Graph (CPG) for your Scala application using
 
 ```scala
-sl ocular
+ocular> createCpg(<inputPaths>)
 ```
 
-### Next Steps
+where `<inputPaths>` is the path of the target application; multiple applications are separated by a comma. For Scala, the path is the archive (JAR or WAR file). For example, `createCpg("subjects/hello-shiftleft-0.0.1-SNAPSHOT.jar").`
```
Comment (Ferada) [inline languages/scala.md]:
```suggestion
where `<inputPaths>` is the path of the target application; multiple applications are separated by a comma. For Scala, the path is the archive (JAR, WAR or EAR file). For example, `createCpg("subjects/hello-shiftleft-0.0.1-SNAPSHOT.jar").`
```

File: languages/golang.md [hunk 8]
```diff
@@ -30,17 +30,18 @@ where
 
 ## ShiftLeft Ocular for Golang
 
-After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md) and [authenticating](../using-cli/authenticating.md), use this command to run ShiftLeft Ocular
+After [installing the ShiftLeft Command Line Interface (CLI)](../using-cli/install-cli.md), [authenticating](../using-cli/authenticating.md) and [starting ShiftLeft Ocular](../using-ocular/getting-started/starting.md), [create the Code Property Graph (CPG)](../using-ocular/getting-started/create-cpg.md) for your Golang application using
 
 ```scala
-sl ocular
+ocular> createCpg(<inputPaths> "GOLANG")
```
[the remark this answers is not in the document]
Comment (NWeber-SL) [reply languages/golang.md]: Done

File: languages/golang.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (NWeber-SL) [reply languages/golang.md]: Removed text about multiple applications

File: languages/java.md (hunk 4 above)
Comment (NWeber-SL) [reply languages/java.md]: Text added

File: languages/java.md (hunk 5 above)
Comment (NWeber-SL) [reply languages/java.md]: Fixed

File: languages/llvm.md (hunk 6 above)
Comment (NWeber-SL) [reply languages/llvm.md]: Fixed

File: languages/scala.md (hunk 7 above)
Comment (NWeber-SL) [reply languages/scala.md]: Text added.

File: languages/llvm.md (hunk 2 above)
Comment (AlexDenisov) [reply languages/llvm.md]: Strictly speaking, this does not work at the moment simply because llvm paty is not integrated into ocular.
But in general, it sounds about right.

We can refine it further as soon as we have ocular integration in place.

@NWeber-SL thank you for putting the things together here 👍
